#include <iostream>
#include <stdexcept> 

double divisao(double num1, double num2) { 
    if (num2 == 0) {
        throw std::invalid_argument("Erro: Divisão por zero!");
    }
    return num1 / num2;
}

int main() {
    double numero1, numero2;

    std::cout << "Digite o numerador: ";
    std::cin >> numero1;

    std::cout << "Digite o denominador: ";
    std::cin >> numero2;

    try {
        std::cout << "O resultado da divisão é: " << divisao(numero1, numero2) << std::endl;
    } catch (const std::invalid_argument& e) {
        std::cerr << e.what() << std::endl;
    }

    return 0;
}
